src/signing.rs

```diff
@@ -3,18 +3,44 @@ use std::time::SystemTime;
 
 use crate::{ChainIndex, Hash256, HexParseError};
 use ed25519_dalek::{Signature as ED25519Signature, Signer, SigningKey, Verifier, VerifyingKey};
-use serde::Serialize;
+use serde::{de::Error, Deserialize, Serialize};
 
 /// An ed25519 public key that can be used to verify a signature
 #[derive(Debug, PartialEq, Clone, Copy)]
 pub struct PublicKey([u8; 32]);
 
+impl PublicKey {
+    const PREFIX: &'static str = "ed25519:";
```

Comment:
I'd like to draw some attention to this. Strictly speaking `types.PublicKey` in `core` adds this prefix when json marshaling. Unfortunately though, `UnlockKey` with specifier ed25519 doesn't serialize the same as `PublicKey` so `UnlockKey` can't replace `PublicKey`.